File: apps/js/src/model/order.ts
export type OrderItem = {
	id: number;
	id_product: number;
	product_name: string;
	unit_price_idr: number;
	quantity: number;
};

export type Order = {
	id: number;
	created_at: string;
	status: string;
	payment_method: string;
	promo_code: string;
	discount_idr: number;
	subtotal_idr: number;
	ship_cost_idr: number;
	total_idr: number;
	ship_name: string;
	ship_phone: string;
	ship_email: string;
	ship_address: string;
	ship_city: string;
	ship_province: string;
	ship_postal_code: string;
	ship_method: string;
	ship_note: string;
	items: OrderItem[];
};

export type OrderRequest = {
	id_address: number;
	payment_method: string;
	ship_method: string;
	promo_code: string;
	ship_note: string;
};

File: apps/go/internal/model/order.go
package model

type OrderItem struct {
	ID           int64  `json:"id" binding:"required"`
	IDProduct    int64  `json:"id_product" binding:"required"`
	ProductName  string `json:"product_name" binding:"required"`
	UnitPriceIdr int64  `json:"unit_price_idr" binding:"required"`
	Quantity     int    `json:"quantity" binding:"required"`
} // @name OrderItem

type Order struct {
	ID             int64       `json:"id" binding:"required"`
	CreatedAt      string      `json:"created_at" binding:"required"`
	Status         string      `json:"status" binding:"required"`
	PaymentMethod  string      `json:"payment_method" binding:"required"`
	PromoCode      string      `json:"promo_code" binding:"required"`
	DiscountIdr    int64       `json:"discount_idr" binding:"required"`
	SubtotalIdr    int64       `json:"subtotal_idr" binding:"required"`
	ShipCostIdr    int64       `json:"ship_cost_idr" binding:"required"`
	TotalIdr       int64       `json:"total_idr" binding:"required"`
	ShipName       string      `json:"ship_name" binding:"required"`
	ShipPhone      string      `json:"ship_phone" binding:"required"`
	ShipEmail      string      `json:"ship_email" binding:"required"`
	ShipAddress    string      `json:"ship_address" binding:"required"`
	ShipCity       string      `json:"ship_city" binding:"required"`
	ShipProvince   string      `json:"ship_province" binding:"required"`
	ShipPostalCode string      `json:"ship_postal_code" binding:"required"`
	ShipMethod     string      `json:"ship_method" binding:"required"`
	ShipNote       string      `json:"ship_note" binding:"required"`
	Items          []OrderItem `json:"items" binding:"required"`
} // @name Order

type OrderRequest struct {
	IDAddress     int64  `json:"id_address" binding:"required"`
	PaymentMethod string `json:"payment_method" binding:"required"`
	ShipMethod    string `json:"ship_method" binding:"required"`
	PromoCode     string `json:"promo_code"`
	ShipNote      string `json:"ship_note"`
} // @name OrderRequest
